edk2toolext/environment/shell_environment.py

```diff
@@ -344,15 +349,18 @@ def get_shell_var(self, var_name):
         return self.active_environ.get(var_name, None)
 
     def set_shell_var(self, var_name, var_data):
-        """Sets the shell variable.
+        """Sets the variable as an OS environment variable.
 
         Args:
             var_name (str): variable to set the value for
             var_data (obj): data to set
 
-        The variable is set both in the `VarDict` and in the OS
+        Raises:
+            (ValueError): var_data is None
         """
         # Check for the "special" shell vars.
+        if var_data is None:
```

Comment:
It would be nice if the exception path was covered by a simple unit test.

Reply:
Test Added. Thanks!